Documentation/gettingstarted/encryption-wireguard.rst

```diff
@@ -0,0 +1,111 @@
+********************************
+
+This guide explains how to configure Cilium with transparent encryption of
+traffic between Cilium-managed endpoints using `Wireguard <https://www.wireguard.com/>`_.
+
+When Wireguard is enabled in Cilium, the agent running on each cluster node
+will establish a secure Wireguard tunnel between it and all other known nodes
+in the cluster. Each node automatically creates its own encryption key-pair and
+distributes its public key via the ``io.cilium.network.wg-pub-key`` annotation in
+the Kubernetes ``CiliumNode`` custom resource object. Each node's public key is then
+used by other nodes to decrypt and encrypt traffic from and to Cilium-managed
+endpoints running on that node.
+
+The Wireguard tunnel endpoint is exposed on UDP port ``51871`` on each node. If
+you run Cilium in an environment that requires firewall rules to enable
+connectivity, you will have to ensure that all Cilium cluster nodes can reach
+each other via that port.
+
+Enable Wireguard in Cilium
+==========================
+
+Before you enable Wireguard in Cilium, please ensure that the Linux distribution
+running on your cluster nodes has support for Wireguard in kernel mode
+(i.e. ``CONFIG_WIREGUARD=m`` on Linux 5.6 and newer, or via the out-of-tree
+Wireguard module on older kernels).
+
+.. tabs::
+
+    .. group-tab:: Cilium CLI
+
+       If you are deploying Cilium with the Cilium CLI, pass the following
+       options:
+
+       .. code-block:: shell-session
+
+          cilium install --config enable-wireguard=true --config enable-l7-proxy=false
+
+    .. group-tab:: Helm
+
+       If you are deploying Cilium with Helm by following
+       :ref:`k8s_install_helm`, pass the following options:
+
+       .. parsed-literal::
+
+           helm install cilium |CHART_RELEASE| \\
+             --namespace kube-system \\
+             --set l7Proxy=false \\
+             --set wireguard.enabled=true
+
+Wireguard may also be enabled manually by setting setting the
+``enable-wireguard: true`` option in the Cilium ``ConfigMap`` and restarting
+each Cilium agent instance.
+
+.. note::
+
+    Wireguard support in Cilium currently lacks support of the following features,
+    which may be resolved in upcoming Cilium releases:
+
+    - Only traffic between two Cilium-managed endpoints (i.e. pod to pod traffic)
+      is encrypted. Traffic between a Cilium-managed pod and a remote host,
+      or traffic between two hosts running the Cilium agent will currently
+      not be encrypted.
+    - L7 policy enforcement and visibility
+    - eBPF-based host routing
+    - Support for older kernels via user-mode Wireguard
+
+   The current status of these limitations is tracked in `#15462 <https://github.com/cilium/cilium/issues/15462>`_.
+
+
+Validate the Setup
+==================
+
+Run a ``bash`` shell in one of the Cilium pods with ``kubectl -n <k8s namespace>
+exec -ti ds/cilium -- bash`` and execute the following commands:
+
+1. Install tcpdump
```

Comment:
IMHO, let's get rid of this `Install tcpdump`. I assume that if a user wants to enable wireguard, they will have enough knowledge how to install tcpdump. Also, it's Ubuntu / Debian specific, and not all users run those distros.

Reply:
This is inside the cilium pod, where we actuall are using a a Debian base image. For managed k8s, the user might not even have access to the node.

Reply:
Aha! Then makes sense.